.circleci/config.yml

```diff
@@ -2,10 +2,76 @@ version: 2.1
 
 orbs:
   python: circleci/python@0.2.1
-  gcp-cloud-run: circleci/gcp-cloud-run@1.0.2
+  docker: circleci/docker@1.2.1
   gcp-gke: circleci/gcp-gke@1.0.4
   kubernetes: circleci/kubernetes@0.11.0
 
+commands:
+  build-and-publish:
+    description: Build and publish a single image
+
+    parameters:
+      tag:
+        type: string
+      path:
+        type: string
+      extra_build_args:
+        type: string
+        default: ""
+      suffix:
+        type: string
+        default: ""
+
+    steps:
+      - docker/build:
+          image: ${DOCKER_IMAGE}
+          tag: <<parameters.tag>><<parameters.suffix>>
+          dockerfile: <<parameters.path>>/Dockerfile
+          extra_build_args: <<parameters.extra_build_args>>
+          step-name: build crlite<<parameters.suffix>> container
+      - docker/push:
+          image: ${DOCKER_IMAGE}
+          tag: <<parameters.tag>><<parameters.suffix>>
+          step-name: push crlite<<parameters.suffix>> container
+
+  publish-docker-pods:
+    description: Build and publish the CRLite containers
+
+    parameters:
+      tag-param:
+        default: ${CIRCLE_SHA1}
+        description: |
+          the environment variable to use as a tag
+        type: string
+
+    steps:
+      - docker/check
+      - build-and-publish:
+          path: containers
+          tag: <<parameters.tag-param>>
+      - docker/update-description:
+          image: ${DOCKER_IMAGE}
```

Comment:
@smarnach Experimentation indicates that these lines require DOCKER_PASSWORD to be an actual user password, not a service key. Which will be used for the ops configuration?